src/figuresClasses.ts

```diff
@@ -1,11 +1,106 @@
-export interface Figure {}
+type Colors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea: () => number;
+}
+
+export abstract class BasicFigure implements Figure {
+  shape: Shapes;
+
+  color: Colors;
+
+  constructor(shape: Shapes, color: Colors) {
+    this.color = color;
+    this.shape = shape;
+  }
+
+  abstract getArea(): number;
+}
+
+export class Triangle extends BasicFigure {
+  constructor(
+    color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier in TypeScript, it implicitly creates a property on the class. Therefore, you don't need to manually assign the parameters to class properties in the constructor for `a`, `b`, and `c`.

GOOD EXAMPLE:
constructor(color: Colors, public a: number, public b: number, public c: number) {
  super(Shapes.Triangle, color);
}